### PR TITLE
SDK-17 -- Add SDK version declaration to the Android SDK

### PR DESCRIPTION
**Pull request by apeterson-branch**
## Reference
SDK-17 -- Add SDK version declaration to the Android SDK

## Description
Google proposed an SDK version declaration format to warn developers who are using vulnerable/old versions of SDKs in their apps.

The declaration should look like this:

  String foobar=”!SDK-VERSION-STRING!:com.example:analytics:beta:3.2.21.0”;

## Testing Instructions
* Sample app should continue to function.
* Decompile the SDK Sample and observe that the string does in fact live in the sources.
* If you enable test mode, we are now printing this tag to the logs.  This is the mechanism for ensuring that the string is not obfuscated away.

My decompile looks like this:

  ./io/branch/referral/Branch.java:    private static final String GOOGLE_VERSION_TAG = "!SDK-VERSION-STRING!:io.branch.sdk.android:library:3.0.2";

## Risk Assessment [`LOW`]
This is just a static string.

- [X] I, the PR creator, have tested — integration, unit, or otherwise — this code.

## Reviewer Checklist (To be checked off by the reviewer only)

- [x] JIRA Ticket is referenced in MR title
- Correctness & Style
    - [x] Conforms to [Style Guides](https://google.github.io/styleguide/javaguide.html)
    - [x] Mission critical pieces are documented in code and out of code as needed
- [ ] Unit Tests reviewed and test issue sufficiently
- [ ] Functionality was reviewed in QA independently by another engineer on the team (clone for each required reviewer)

### CODE REVIEW

File: Branch-SDK-TestBed/build.gradle [hunk 1]
```diff
@@ -3,7 +3,7 @@ apply plugin: 'com.android.application'
 dependencies {
     implementation project(':Branch-SDK')
     //debugImplementation 'com.squareup.leakcanary:leakcanary-android:1.+'
-    implementation ('com.google.android.gms:play-services-ads:10.0.1')
+    implementation ('com.google.android.gms:play-services-ads:16.0.0')
```
Comment (csalmi-branch) [inline Branch-SDK-TestBed/build.gradle]: It seems like that version had some issues, is this something you are concerned with?
"Known Issue with October 2, 2018 release
Some of the libraries in the October 2, 2018 release contained an issue in their POM dependencies. This most likely manifests as dex merge issues when some dependency version is updated (even when using the strict-version-matcher-plugin). To resolve, update to the Recommended Minimum Version as follows:"

Comment (csalmi-branch) [reply Branch-SDK-TestBed/build.gradle]: Also, is there a good way to prevent these dependencies from getting a year and a half out of date?

Comment (apeterson-branch) [reply Branch-SDK-TestBed/build.gradle]: Note that this was a dependency in the sample app, not the SDK.  The sample wouldn't compile without making the change.

Comment (apeterson-branch) [reply Branch-SDK-TestBed/build.gradle]: Updating to 17.x.x causes a crash in the Sample app.  I think 16.x.x is as high as we can go here.

**Pull request merged**